membership-common/src/main/scala/com/gu/memsub/subsv2/services/CatalogService.scala

```diff
@@ -36,21 +40,35 @@ object FetchCatalog {
 
 object CatalogService extends SafeLogging {
 
-  def read[M[_]: Functor](fetchCatalog: M[String \/ JsValue]): M[CatalogMap] = {
+  def read[M[_]: Functor](fetchCatalog: M[String \/ JsValue], productIds: ProductIds): M[Catalog] =
+    for {
+      failableJsCatalog <- fetchCatalog
+    } yield {
 
-    val catalog: M[String \/ CatalogMap] = {
-      fetchCatalog.map(_ => ())
-      for {
-        failableCatalog <- fetchCatalog
-      } yield for {
-        catalog <- failableCatalog
-        plans <- Json.fromJson[List[CatalogZuoraPlan]](catalog).asEither.toDisjunction.leftMap(_.toString)
+      val failableCatalog = for {
+        catalog <- failableJsCatalog
+        plans <- Json.fromJson[List[ProductRatePlan]](catalog)(productsReads).asEither.toDisjunction.leftMap(_.toString)
       } yield plans.map(catalogZuoraPlan => catalogZuoraPlan.id -> catalogZuoraPlan).toMap
+
+      val catalogMap = failableCatalog
+        .leftMap[CatalogMap](error => {
+          logger.errorNoPrefix(scrub"error: $error"); Map.empty // not sure why we empty-on-failure
+        })
+        .merge
+
+      Catalog(catalogMap ++ patronPlans, productIds)
     }
 
-    catalog.map(_.leftMap[CatalogMap](error => {
-      logger.errorNoPrefix(scrub"error: $error"); Map.empty
-    }).merge)
+  private val patronPlans = Map[ProductRatePlanId, ProductRatePlan](
+    Catalog.guardianPatronProductRatePlanId -> ProductRatePlan(
+      Catalog.guardianPatronProductRatePlanId,
+      "Guardian Patron", // was subscription.plan.id but isn't used for patron
+      SubsV2ProductIds.guardianPatronProductId,
+      Map[ProductRatePlanChargeId, ProductRatePlanChargeProductType](
+        Catalog.guardianPatronProductRatePlanChargeId -> ProductRatePlanChargeProductType.GuardianPatron,
+      ),
+      Some(ProductType("Membership")), // not used for patron - only used for payment related emails
+    ),
+  )
```

Comment:
this is where we add the fake patron plans to the in-memory catalog.  This means they can be looked up as normal by the controllers etc (rather than trying to shoehorn eithers or coproduct types everywhere)